File: src/main/java/yunas/db/DefaultDataSourceStrategy.java
package yunas.db;

import yunas.configuration.Configuration;
import yunas.exception.YunasExceptionProvider;
import yunas.util.BaseUtil;

/**
 * DefaultDataSourceStrategy.
 *
 * Choose a connection pool.
 *
 */
public class DefaultDataSourceStrategy implements DataSourceStrategy {

     @Override
     public ConnectionPoolFactory  getFactory(Configuration conf) {

        if (BaseUtil.blank(conf.get("yunas.cp")) || BaseUtil.equalsIgnorecase(conf.get("yunas.cp"), "Hicaricp")) {
            return new HikariCPFactory();
        }

        // Add other CP
        throw new YunasExceptionProvider().unknownConnectionPoolName();
    }
}

File: src/main/kotlin/yunas/db/DefaultDataSourceStrategy.kt
package yunas.db

import yunas.configuration.Configuration
import yunas.exception.YunasExceptionProvider
import yunas.util.BaseUtil

/**
 * DefaultDataSourceStrategy.
 *
 * Choose a connection pool.
 *
 */
class DefaultDataSourceStrategy : DataSourceStrategy {

    override fun getFactory(conf: Configuration): ConnectionPoolFactory {

        if (BaseUtil.blank(conf.get("yunas.cp")) || BaseUtil.equalsIgnoreCase(conf.get("yunas.cp"), "Hicaricp")) {
            return HikariCPFactory()
        }

        // Add other CP
        throw YunasExceptionProvider().unknownConnectionPoolName()
    }
}
